### ISSUE TITLE
High CPU usage in charts.d.plugin on armhf

### ISSUE THREAD

**Issue by stulluk** [firehol/netdata#169]
Hi, 
Thank you soo much for this awesome app. Unfortunately, I am having a trouble on my debian7 wheezy armhf board with netdata.

charts.d.plugin allways use %100 cpu when I start netdata.

Here is screenshot of my htop:  http://postimg.org/image/j3igzk4gp/

Here is my rc.local file:
mkdir -p /var/log/netdata
echo 1 >/sys/kernel/mm/ksm/run
echo 1000 >/sys/kernel/mm/ksm/sleep_millisecs
/usr/sbin/netdata -pidfile /var/run/netdata.pid

exit 0

Here is what happens when I strace that PID:
root@i3-e0ca ~ # strace -p 547
...
...
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_BLOCK, NULL, [], 8)  = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, [CHLD], [], 8) = 0
rt_sigprocmask(SIG_SETMASK, [], ^C <unfinished ...>
Process 547 detached
root@i3-e0ca ~ # 

Any ideas?

**Comment by ktsaou** [comment]
stop netdata and run charts.d.plugin by hand, like this:

```sh
/usr/libexec/netdata/plugins/charts.d.plugin debug 10
```

This will give you a lot of debug output. the plugin will attempt to collect data every 10 seconds.
Can you post the output?

**Comment by stulluk** [comment]
Hi, thank you very much for the response.

Here is the debug output of the plugin: http://paste.ubuntu.com/15613405/

Pls note that, if I start the plugin standalone, it does not consume %100 cpu...

**Comment by ktsaou** [comment]
Most probably a charts.d.plugin is causing this. This is how to trace it:

Edit `/etc/netdata/charts.d.conf` and set:

```sh
enable_all_charts="no"
X="yes"
```

When `X` is a plugin (you can find their names at the top of the pastebin you posted).

Please enable one by one, until you have the offending one.

Of course, restart netdata to apply the changes, each time.

**Comment by stulluk** [comment]
Thank you soo much for your answer. I checked all plugins from the log file, added to charts.d.conf, and wrote "no" to every single of them.

For every tries, I rebooted my embedded system.

Unfortunately, here was the result: http://postimg.org/gallery/1pmwq2dwe/95c1206d/

I decided to write "What else can I do?" message, but suddenly something strange happened. CPU usage went back to normal !

After that, I keep continue searching. Suddenly I saw something strange, when I set 

enable_all_charts=no

then, sensors and example charts disappeared, but I have some other charts here. Here is the comparison:

http://postimg.org/gallery/1sjgt89uw/22854551/

So, when I set "no" , after bootup, htop shows %100 cpu usage for one core, and  around 1minute 41 seconds later, that usage drops to %3-%6 , and system becomes normal.

( reboot 3-4 times to check behaviour, allways the same )

So, except sensors, everything seems working fine under debian wheezy 7 on Allwinner A20 based Hardware.

Thank you so much for this awesome app.

issue can be marked as solved.
